File: PACMAN-FILTER/Graph/ScriptTemplate/Or/CGOr.lua
local CGOr = CGOr or {}
CGOr.__index = CGOr

function CGOr.new()
    local self = setmetatable({}, CGOr)
    self.inputs = {}
    return self
end

function CGOr:setInput(index, func)
    self.inputs[index] = func
end

function CGOr:getOutput(index)
    return self.inputs[0]() or self.inputs[1]()
end

return CGOr

File: PACMAN-FILTER/Graph/ScriptTemplate/Or/CGOr.js
/**
 * @file CGOr.js
 * @author
 * @date 2021/8/15
 * @brief CGOr.js
 * @copyright Copyright (c) 2021, ByteDance Inc, All Rights Reserved
 */

const {BaseNode} = require('./BaseNode');
const Amaz = effect.Amaz;

class CGOr extends BaseNode {
  constructor() {
    super();
  }

  getOutput() {
    return this.inputs[0]() || this.inputs[1]();
  }
}

exports.CGOr = CGOr;
